### PR TITLE
Fix Restore + Ready for Level 7 Package_check

### PR DESCRIPTION
**Pull request by yalh76**

### CODE REVIEW

**Comment by yalh76** [comment]
This one will be the good one ^^
I'll will work on another branch

wekan

**Comment by yalh76** [comment]
This pull request can be merged ^^
The other [PR ](https://github.com/YunoHost-Apps/wekan_ynh/pull/31) is more cosmetic.

File: scripts/restore [hunk 1]
```diff
@@ -63,18 +64,18 @@ ynh_install_app_dependencies "mongodb mongodb-server"
...
-chown -R $app $final_path
-pushd $final_path/bundle/programs/server
-ynh_use_nodejs
-npm install
-popd
+#chown -R $app $final_path
+#pushd $final_path/programs/server
+#ynh_use_nodejs
+#npm install
+#popd
```
Comment (alexAubin) [inline scripts/restore]: So eeeh, don't we need those ? Or is it that the whole `npm install` deps are backuped somehow ? (I dunno if we want to do this or not)

File: scripts/restore [hunk 2]
```diff
@@ -106,5 +107,12 @@ fi
...
 #=================================================
-systemctl restart $app
+
 systemctl reload nginx
+
+#=================================================
+# START SERVICE
+#=================================================
+
+ynh_systemd_action --action=start --service_name=$app --log_path="systemd"
```
Comment (alexAubin) [inline scripts/restore]: No `--line_match` here ? Is that voluntary ? :stuck_out_tongue:

File: scripts/upgrade [hunk 3]
```diff
@@ -113,5 +114,12 @@ popd
...
 #=================================================
-systemctl restart $app
+
 systemctl reload nginx
+
+#=================================================
+# START SERVICE
+#=================================================
+
+ynh_systemd_action --action=start --service_name=$app --log_path="systemd"
```
Comment (alexAubin) [inline scripts/upgrade]: (same question about `--line_match`)

File: scripts/restore (hunk 1 above)
Comment (yalh76) [reply scripts/restore]: yes, all modules are located in the /var/www/wekan/node_modules folder. So they are backup and restore with the whole content of /var/www/wekan

File: scripts/restore (hunk 2 above)
Comment (yalh76) [reply scripts/restore]: when the system start after restore: here are the log in journalctl
wekan
In my opinion, nothing helpful to define that wekan is fully started.
But I can add: --line_match="Started Wekan"
Let me know

File: scripts/upgrade (hunk 3 above)
Comment (yalh76) [reply scripts/upgrade]: like restore, same response ^^

File: scripts/restore (hunk 1 above)
Comment (alexAubin) [reply scripts/restore]: Hmmmmokay, that is weird ... I don't know if that behavior was already there before your PR (I think I checked that and it's not from your PR ?) but naively it looks to me like it's not a good idea to have all those /node_modules being backuped ? Those are just dependencies (so not "data and conf" strictly speaking) and might weight *a lot*.

Like, I installed a ~small npm-based utility on my laptop and now I have a ~400 Mo node_modules folder which sounds crazy to me.

What do you think ?

(Granted that it's not a regression and that npm install is probably useless / a waste of time if node_modules are already there - so we can deal with this later - but just to try to clarify what we should do with this :wink: )

Comment (yalh76) [reply scripts/restore]: Nop, that was not from my PR, already there in the master branch when doing: `npm install`

For backuping the /node_modules, I partially agree with you:

1. I's weird to backup so much modules, but after installation done, I don't know what part of the /node_modules need to be kept or what part can be cleaned up.
2. NodeJS seems to evolve very quickly and modules too... And as far as I've seen (I'm not developer at all, more Windows AdminSys and I don't know javascript/NodeJ ), NodeJS modules seems to be broken sometimes. So, be able to restore the full working environment without downloading new versions of modules (some maybe broken), make me more comfortable/secure

And yes node_modules are just dependencies, nodejs core seems very small, so they develop and maintain a lot of modules. For example, one to see if a number is even or odd. one other module to check if a number is a multiple of 10, etc....

They also have different modules that do the same thing... A friend of mine is a nodejs developer, and speaking of nodejs developpment, he said to me that the tricky part of nodejs developpment is not developing, is to choose the good module: one still maintain, up to date, etc...

Maybe we can do without the npm install in the restore script but I really don't know if it only download modules... it seems to also "compile" things....

**Comment by alexAubin** [comment]
(Bumpz @zamentur in case you have time and thoughts on this)

File: scripts/upgrade (hunk 3 above)
Comment (alexAubin) [inline scripts/upgrade]: So eh, sorry if I already asked this somewhere else .. but shouldn't be a restart here ?

Comment (yalh76) [reply scripts/upgrade]: yes better to user restart than start, as service wekan has not been stoped

**Comment by alexAubin** [comment]
Meeerging o/ 

Thanks for your work and sorry for the delay in reviewing / merging :sweat_smile:

**Comment by yalh76** [comment]
> 
> 
> Meeerging o/
> 
> Thanks for your work and sorry for the delay in reviewing / merging 😅

No problem ^^.
I don't know if you have seen the other [PR ](https://github.com/YunoHost-Apps/wekan_ynh/pull/31) about applying last example_ynh ? maybe I need to redo it as there is a new version of [example_ynh](https://github.com/YunoHost/example_ynh) since 3 days...

**Pull request merged**